src/Transports/MassTransit.AmazonSqsTransport/AmazonSqsTransport/Middleware/ConfigureAmazonSqsTopologyFilter.cs

```diff
@@ -72,7 +72,11 @@ async Task ConfigureTopology(ClientContext context)
         await Task.WhenAll(queues).ConfigureAwait(false);
 
         IEnumerable<Task> subscriptions = _brokerTopology.QueueSubscriptions.Select(subscription => Declare(context, subscription));
-        await Task.WhenAll(subscriptions).ConfigureAwait(false);
+
+        foreach (var subscription in subscriptions)
+        {
+            await subscription.ConfigureAwait(false);
+        }
```

Comment:
This may have a negative impact on bus startup time, since it needs to check for the subscriptions and update them sequentially.

Reply:
Right, but there was also a race condition when multiple threads tried to update the same policy. From 20 statements most of them were lost. Do you have any ideas/suggestions on how to deal with this?

Reply:
This would only be on a single topic, right? 

1. Two receive endpoints consuming the same topic, should lock the topic for updates at the connection level to avoid the race condition.
2. One receive endpoint with two consumers same topic should not add two conditions, the broker topology should identify the duplicate and only have one condition.
3. If the conditions already exist/match on the broker, don't update
4. If the receive endpoint adds a new condition updating the original, it shouldn't create a new policy just add the condition (the two receive endpoints problem again).